src/components/app/data/hooks/useContentDiscoveryNavLink.jsx

```diff
@@ -0,0 +1,29 @@
+import { NavLink } from 'react-router-dom';
+import { FormattedMessage } from '@edx/frontend-platform/i18n';
+import useAcademies from './useAcademies';
+import useEnterpriseCustomer from './useEnterpriseCustomer';
+
+export default function useContentDiscoveryNavLink(mainMenuLinkClassName) {
```

Comment:
Given this hook will only used by `SiteHeaderNavMenu`, I would recommend not defining it in the more "global" hooks location (app/data/hooks), which are intended to be broadly re-used across the application.

Could this hook be defined within the `site-header` module itself since it's only relevant for that module?